make/xml.cmake

```diff
@@ -37,3 +37,54 @@ function(XML_LINT)
 
 endfunction(XML_LINT)
 
+function(XML_SORT)
+  # ~~~
+  # XML_SORT(
+  # NAME
+  # FILE
+  # OUTPUT
+  # )
+  set(oneValueArgs NAME FILE OUTPUT)
+  cmake_parse_arguments(
+    XML_SORT
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+    )
+
+  set(XML_SORT_XSL ${symbiflow-arch-defs_SOURCE_DIR}/common/xml/xmlsort.xsl)
+
+  get_file_location(XML_SORT_INPUT_LOCATION ${XML_SORT_FILE})
+
+  get_file_target(XML_SORT_INPUT_TARGET ${XML_SORT_FILE})
+  get_target_property(INCLUDE_FILES ${XML_SORT_INPUT_TARGET} INCLUDE_FILES)
+  set(DEPS "")
+  append_file_dependency(DEPS ${XML_SORT_FILE})
+
+  get_target_property_required(XSLTPROC env XSLTPROC)
+  get_target_property(XSLTPROC_TARGET env XSLTPROC_TARGET)
+
+  add_custom_command(
+    OUTPUT ${XML_SORT_OUTPUT}
+    DEPENDS
+      ${XML_SORT_XSL}
+      ${XML_SORT_INPUT_LOCATION}
+      ${XML_SORT_INPUT_TARGET}
+      ${DEPS}
+      ${XSLTPROC} ${XSLTPROC_TARGET}
+    COMMAND
+      ${XSLTPROC}
+      --nomkdir
+      --nonet
+      --xinclude
+      --output ${CMAKE_CURRENT_BINARY_DIR}/${XML_SORT_OUTPUT}
+      ${XML_SORT_XSL}
+      ${XML_SORT_INPUT_LOCATION}
+  )
+  add_file_target(FILE ${XML_SORT_OUTPUT} GENERATED)
```

Comment:
We should set INCLUDE_FILES property for the XML_SORT_OUTPUT target

add_file_target with GENERATED flag will not add this automatically, so we need to handle this here

Reply:
I do think that in this case it is fine to leave it as it is as there shouldn't be files to be included in the `XML_SORT_OUTPUT` dependencies.

Reply:
makes sense